### PR TITLE
build: show UID and GID in welcome message

### PR DESCRIPTION
**Pull request by scarf005**
# Motivation

image

continuation of #8445 after further discussion in https://github.com/NixOS/nix/pull/8445#issuecomment-1590327217

# Context

may help reduce confusions with UIDs like in
- #7068
- https://github.com/NixOS/nixpkgs/issues/235546

# Checklist for maintainers

Maintainers: tick if completed or explain if not relevant

 - [ ] agreed on idea
 - [ ] agreed on implementation strategy
 - [ ] tests, as appropriate
   - functional tests - `tests/**.sh`
   - unit tests - `src/*/tests`
   - integration tests - `tests/nixos/*`
 - [ ] documentation in the manual
 - [ ] documentation in the internal API docs
 - [ ] code and comments are self-explanatory
 - [ ] commit message explains why the change was made
 - [ ] new feature or incompatible change: updated release notes

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: scripts/install-multi-user.sh [hunk 1]
```diff
@@ -700,6 +700,10 @@ EOF
 }
 
 welcome_to_nix() {
+    local -r NIX_UID_RANGES="${NIX_FIRST_BUILD_UID} ~ $((NIX_FIRST_BUILD_UID + NIX_USER_COUNT - 1))"
```
Comment (Ericson2314) [inline scripts/install-multi-user.sh:703]: I am not familiar with `~` for `[low..high)` ranges. Is it standard?

Comment (scarf005) [reply scripts/install-multi-user.sh:703]: I think it's [CJK specific](https://en.wikipedia.org/wiki/Tilde#Range). In this case, i used it to denote `[low..high]` range (i.e `{30001,30002, ... 30032}`). Should i change the format to something like `[30001..30032]`?

Comment (Ericson2314) [reply scripts/install-multi-user.sh:703]: Thanks for the background info! Yeah I think that might be better. `[...]` or `[..,)` is fine with me as I suppose we don't need to worry about the 0-build-user case :).

Comment (scarf005) [reply scripts/install-multi-user.sh]: alright, committed and updated screenshot.

File: scripts/install-multi-user.sh [hunk 2]
```diff
@@ -700,8 +700,8 @@ EOF
 }
 
 welcome_to_nix() {
-    local -r NIX_UID_RANGES="${NIX_FIRST_BUILD_UID} ~ $((NIX_FIRST_BUILD_UID + NIX_USER_COUNT - 1))"
-    local -r RANGE_TEXT=$(echo -ne "${BLUE}(uid ${NIX_UID_RANGES})${ESC}")
+    local -r NIX_UID_RANGES="${NIX_FIRST_BUILD_UID}..$((NIX_FIRST_BUILD_UID + NIX_USER_COUNT - 1))"
+    local -r RANGE_TEXT=$(echo -ne "${BLUE}(uid [${NIX_UID_RANGES}])${ESC}")
```
Comment (Ericson2314) [inline scripts/install-multi-user.sh:704]:
```suggestion
    local -r RANGE_TEXT=$(echo -ne "${BLUE}(uids [${NIX_UID_RANGES}])${ESC}")
```

**Review by Ericson2314 — APPROVED** [review]
@fricklerhandwerk approved the concept in the other PR, and I am signing off on the details. This is very low-impact maintenance-wise (hopefully higher impact for useres!) so I think that is enough review :).

**Pull request merged**